2019-08-26 13:12:38,580 ERROR: Exception on /edit_profile [POST] [in /Users/darwin/Dropbox/webdev/projects/microblog/venv/lib/python3.7/site-packages/flask/app.py:1891]
Traceback (most recent call last):
  File "/Users/darwin/Dropbox/webdev/projects/microblog/venv/lib/python3.7/site-packages/sqlalchemy/engine/base.py", line 1244, in _execute_context
    cursor, statement, parameters, context
  File "/Users/darwin/Dropbox/webdev/projects/microblog/venv/lib/python3.7/site-packages/sqlalchemy/engine/default.py", line 552, in do_execute
    cursor.execute(statement, parameters)
sqlite3.IntegrityError: UNIQUE constraint failed: user.username

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/darwin/Dropbox/webdev/projects/microblog/venv/lib/python3.7/site-packages/flask/app.py", line 2446, in wsgi_app
    response = self.full_dispatch_request()
  File "/Users/darwin/Dropbox/webdev/projects/microblog/venv/lib/python3.7/site-packages/flask/app.py", line 1951, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/Users/darwin/Dropbox/webdev/projects/microblog/venv/lib/python3.7/site-packages/flask/app.py", line 1820, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/Users/darwin/Dropbox/webdev/projects/microblog/venv/lib/python3.7/site-packages/flask/_compat.py", line 39, in reraise
    raise value
  File "/Users/darwin/Dropbox/webdev/projects/microblog/venv/lib/python3.7/site-packages/flask/app.py", line 1949, in full_dispatch_request
    rv = self.dispatch_request()
  File "/Users/darwin/Dropbox/webdev/projects/microblog/venv/lib/python3.7/site-packages/flask/app.py", line 1935, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/Users/darwin/Dropbox/webdev/projects/microblog/venv/lib/python3.7/site-packages/flask_login/utils.py", line 261, in decorated_view
    return func(*args, **kwargs)
  File "/Users/darwin/Dropbox/webdev/projects/microblog/app/routes.py", line 86, in edit_profile
    db.session.commit()
  File "/Users/darwin/Dropbox/webdev/projects/microblog/venv/lib/python3.7/site-packages/sqlalchemy/orm/scoping.py", line 162, in do
    return getattr(self.registry(), name)(*args, **kwargs)
  File "/Users/darwin/Dropbox/webdev/projects/microblog/venv/lib/python3.7/site-packages/sqlalchemy/orm/session.py", line 1027, in commit
    self.transaction.commit()
  File "/Users/darwin/Dropbox/webdev/projects/microblog/venv/lib/python3.7/site-packages/sqlalchemy/orm/session.py", line 494, in commit
    self._prepare_impl()
  File "/Users/darwin/Dropbox/webdev/projects/microblog/venv/lib/python3.7/site-packages/sqlalchemy/orm/session.py", line 473, in _prepare_impl
    self.session.flush()
  File "/Users/darwin/Dropbox/webdev/projects/microblog/venv/lib/python3.7/site-packages/sqlalchemy/orm/session.py", line 2459, in flush
    self._flush(objects)
  File "/Users/darwin/Dropbox/webdev/projects/microblog/venv/lib/python3.7/site-packages/sqlalchemy/orm/session.py", line 2597, in _flush
    transaction.rollback(_capture_exception=True)
  File "/Users/darwin/Dropbox/webdev/projects/microblog/venv/lib/python3.7/site-packages/sqlalchemy/util/langhelpers.py", line 68, in __exit__
    compat.reraise(exc_type, exc_value, exc_tb)
  File "/Users/darwin/Dropbox/webdev/projects/microblog/venv/lib/python3.7/site-packages/sqlalchemy/util/compat.py", line 153, in reraise
    raise value
  File "/Users/darwin/Dropbox/webdev/projects/microblog/venv/lib/python3.7/site-packages/sqlalchemy/orm/session.py", line 2557, in _flush
    flush_context.execute()
  File "/Users/darwin/Dropbox/webdev/projects/microblog/venv/lib/python3.7/site-packages/sqlalchemy/orm/unitofwork.py", line 422, in execute
    rec.execute(self)
  File "/Users/darwin/Dropbox/webdev/projects/microblog/venv/lib/python3.7/site-packages/sqlalchemy/orm/unitofwork.py", line 589, in execute
    uow,
  File "/Users/darwin/Dropbox/webdev/projects/microblog/venv/lib/python3.7/site-packages/sqlalchemy/orm/persistence.py", line 236, in save_obj
    update,
  File "/Users/darwin/Dropbox/webdev/projects/microblog/venv/lib/python3.7/site-packages/sqlalchemy/orm/persistence.py", line 996, in _emit_update_statements
    statement, multiparams
  File "/Users/darwin/Dropbox/webdev/projects/microblog/venv/lib/python3.7/site-packages/sqlalchemy/engine/base.py", line 988, in execute
    return meth(self, multiparams, params)
  File "/Users/darwin/Dropbox/webdev/projects/microblog/venv/lib/python3.7/site-packages/sqlalchemy/sql/elements.py", line 287, in _execute_on_connection
    return connection._execute_clauseelement(self, multiparams, params)
  File "/Users/darwin/Dropbox/webdev/projects/microblog/venv/lib/python3.7/site-packages/sqlalchemy/engine/base.py", line 1107, in _execute_clauseelement
    distilled_params,
  File "/Users/darwin/Dropbox/webdev/projects/microblog/venv/lib/python3.7/site-packages/sqlalchemy/engine/base.py", line 1248, in _execute_context
    e, statement, parameters, cursor, context
  File "/Users/darwin/Dropbox/webdev/projects/microblog/venv/lib/python3.7/site-packages/sqlalchemy/engine/base.py", line 1466, in _handle_dbapi_exception
    util.raise_from_cause(sqlalchemy_exception, exc_info)
  File "/Users/darwin/Dropbox/webdev/projects/microblog/venv/lib/python3.7/site-packages/sqlalchemy/util/compat.py", line 398, in raise_from_cause
    reraise(type(exception), exception, tb=exc_tb, cause=cause)
  File "/Users/darwin/Dropbox/webdev/projects/microblog/venv/lib/python3.7/site-packages/sqlalchemy/util/compat.py", line 152, in reraise
    raise value.with_traceback(tb)
  File "/Users/darwin/Dropbox/webdev/projects/microblog/venv/lib/python3.7/site-packages/sqlalchemy/engine/base.py", line 1244, in _execute_context
    cursor, statement, parameters, context
  File "/Users/darwin/Dropbox/webdev/projects/microblog/venv/lib/python3.7/site-packages/sqlalchemy/engine/default.py", line 552, in do_execute
    cursor.execute(statement, parameters)
sqlalchemy.exc.IntegrityError: (sqlite3.IntegrityError) UNIQUE constraint failed: user.username
[SQL: UPDATE user SET username=?, about_me=? WHERE user.id = ?]
[parameters: ('susan', 'I am a NYC based Full-stack Developer. I work with HTML, CSS, JavaScript, React, Express, PostgreSQL, Sequelize and more.', 3)]
(Background on this error at: http://sqlalche.me/e/gkpj)
2019-08-26 14:03:52,010 INFO: Microblog startup [in /Users/darwin/Dropbox/webdev/projects/microblog/app/__init__.py:43]
2019-08-26 14:35:05,050 INFO: Microblog startup [in /Users/darwin/Dropbox/webdev/projects/microblog/app/__init__.py:43]
2019-08-26 14:37:37,368 INFO: Microblog startup [in /Users/darwin/Dropbox/webdev/projects/microblog/app/__init__.py:43]
2019-08-26 14:38:55,827 INFO: Microblog startup [in /Users/darwin/Dropbox/webdev/projects/microblog/app/__init__.py:43]
2019-08-26 14:39:45,546 INFO: Microblog startup [in /Users/darwin/Dropbox/webdev/projects/microblog/app/__init__.py:43]
2019-08-26 14:40:09,543 INFO: Microblog startup [in /Users/darwin/Dropbox/webdev/projects/microblog/app/__init__.py:43]
2019-08-26 15:51:46,732 INFO: Microblog startup [in /Users/darwin/Dropbox/webdev/projects/microblog/app/__init__.py:43]
2019-08-26 15:53:42,181 INFO: Microblog startup [in /Users/darwin/Dropbox/webdev/projects/microblog/app/__init__.py:43]
2019-08-26 15:54:21,381 INFO: Microblog startup [in /Users/darwin/Dropbox/webdev/projects/microblog/app/__init__.py:43]
2019-08-26 15:54:53,043 INFO: Microblog startup [in /Users/darwin/Dropbox/webdev/projects/microblog/app/__init__.py:43]
2019-08-27 11:06:32,470 INFO: Microblog startup [in /Users/darwin/Dropbox/webdev/projects/microblog/app/__init__.py:43]
2019-08-27 11:07:08,770 INFO: Microblog startup [in /Users/darwin/Dropbox/webdev/projects/microblog/app/__init__.py:43]
2019-08-27 11:07:21,559 INFO: Microblog startup [in /Users/darwin/Dropbox/webdev/projects/microblog/app/__init__.py:43]
2019-08-27 13:42:10,465 INFO: Microblog startup [in /Users/darwin/Dropbox/webdev/projects/microblog/app/__init__.py:43]
2019-08-27 14:29:55,736 INFO: Microblog startup [in /Users/darwin/Dropbox/webdev/projects/microblog/app/__init__.py:43]
2019-08-27 14:52:24,603 INFO: Microblog startup [in /Users/darwin/Dropbox/webdev/projects/microblog/app/__init__.py:43]
2019-08-27 14:55:18,269 INFO: Microblog startup [in /Users/darwin/Dropbox/webdev/projects/microblog/app/__init__.py:43]
2019-08-27 14:56:17,776 INFO: Microblog startup [in /Users/darwin/Dropbox/webdev/projects/microblog/app/__init__.py:43]
2019-08-27 14:56:40,485 INFO: Microblog startup [in /Users/darwin/Dropbox/webdev/projects/microblog/app/__init__.py:43]
2019-08-27 14:57:51,231 INFO: Microblog startup [in /Users/darwin/Dropbox/webdev/projects/microblog/app/__init__.py:43]
2019-08-27 14:58:11,617 INFO: Microblog startup [in /Users/darwin/Dropbox/webdev/projects/microblog/app/__init__.py:43]
2019-08-27 14:59:12,907 INFO: Microblog startup [in /Users/darwin/Dropbox/webdev/projects/microblog/app/__init__.py:43]
2019-08-27 14:59:26,462 INFO: Microblog startup [in /Users/darwin/Dropbox/webdev/projects/microblog/app/__init__.py:43]
